New Topic: Cloud Computing (#930)
Codecademy/docs: content/cloud-computing/cloud-computing.md

---
Title: 'Cloud Computing'
Description: 'Cloud computing refers to an architecture where computing resources and hardware are maintained and managed remotely by an entity other than the user.'
Codecademy Hub Page: 'https://www.codecademy.com/catalog/subject/cloud-computing'
CatalogContent:
  - 'learn/foundations-of-cloud-computing'
  - 'paths/back-end-engineer-career-path'
---

> **Cloud computing** refers to an architecture where computing resources and hardware are maintained and managed remotely by an entity other than the user. This is becoming popular with businesses and ubiquitous in personal computing. For example, whenever a person sets up a website on GoDaddy, plays World of Warcraft, or streams a movie on Netflix, they are using a cloud computing solution.

Reviewer: > **Cloud computing** refers to an architecture where computing resources

I like this approach of bolding a topic (in addition to a concept or term) 🚀 Let's do this for all new topics going forward! 😄

Reviewer: I made a similar suggestion down in the `go.md` file.

## Advantages

Cloud computing offers the following advantages to the user:

> - **Scalability:** Adding additional capacity is quicker and easier in a cloud environment then in an on-premises data center. When a user manages a data center themselves, additional capacity involves purchasing, installing and configuring new hardware. A cloud solution provides access to many more resources than would otherwise be possible.

Reviewer: ```suggestion
- **Scalability:** Adding additional capacity is quicker and easier in a cloud environment then in an on-premises data center. When a user manages a data center themselves, additional capacity involves purchasing, installing, and configuring new hardware. A cloud solution provides access to many more resources than would otherwise be possible.
```

> - **Scalability:** Adding additional capacity is quicker and easier in a cloud environment then in an on-premises data center. When a user manages a data center themselves, additional capacity involves purchasing, installing, and configuring new hardware. A cloud solution provides access to many more resources than would otherwise be possible.

Reviewer 2: Concern: meaning of on-premises data center may not be clear

- **Security:** A cloud solution often provides more security than a business, especially a small business, can manage on their own.

> - **Availability:** Most cloud solutions have more robust and redundant architecture than a small to medium business can afford, leading to better uptime for users and clients.

Reviewer 2: Concern: meaning of robust and redundant architecture may be unclear. Perhaps something like:

Cloud solutions are able to provide services that are guaranteed to stay available.

- **Performance:** A cloud solution will provide more powerful hardware than a business could afford in its own data center.

> - **Cost:** A cloud solution is typically more affordable than an on-prem solution offering the same capabilities.

Reviewer: ```suggestion
- **Cost:** A cloud solution is typically more affordable than an on-premises solution offering the same capabilities.
```

> Generally, cloud computing removes the necessity of a business to maintain a dedicated data center, with all the hardware and personnel it requires, for a particular solution, allowing the business to allocate those resources to other tasks.

Reviewer 2: Run on sentence. Perhaps:

Cloud computing removes the need for a business to maintain a dedicated data center. This frees the business from the cost of all the personnel and hardware that a data center requires.

---
Title: 'Cloud Computing'
Description: 'Cloud computing refers to an architecture where computing resources and hardware are maintained and managed remotely by an entity other than the user.'
Codecademy Hub Page: 'https://www.codecademy.com/catalog/subject/cloud-computing'
CatalogContent:
  - 'learn/foundations-of-cloud-computing'
  - 'paths/back-end-engineer-career-path'
---

**Cloud computing** refers to an architecture where computing resources and hardware are maintained and managed remotely by an entity other than the user. This is becoming more popular with businesses, and has even become ubiquitous in personal computing. Whenever a person sets up a website on GoDaddy, plays World of Warcraft, or streams a movie on Netflix, they are using a cloud computing solution.

> ## How does Cloud Computing work?

Reviewer: ```suggestion
## How Cloud Computing Works
```

> The cloud computing service is divided into two sections, frontend and backend. They are connected through the Internet. The frontend consists of the client’s computer or computer network. On the backend, the cloud is made of the following:

Reviewer: ```suggestion
The cloud computing service is divided into two sections,[ front-end](https://www.codecademy.com/resources/docs/general/front-end) and [back-end](https://www.codecademy.com/resources/docs/general/back-end), that are connected through the Internet. The front-end consists of the client’s computer or computer network. On the back-end, the cloud is made of the following:
```

- Computers

> - Servers
> - Data Storage Systems

Reviewer: ```suggestion
- [Servers](https://www.codecademy.com/resources/docs/general/server)
- [Data Storage Systems](https://www.codecademy.com/resources/docs/general/database)
```

Was it correct to assume that "data storage systems" pertains to "databases"?

> The cloud computing service is divided into two sections, [front-end](https://www.codecademy.com/resources/docs/general/front-end) and [back-end](https://www.codecademy.com/resources/docs/general/back-end), that are connected through the Internet. The front-end consists of the client’s computer or computer network. On the back-end, the cloud is made of the following:

Reviewer 2: I don't agree that Cloud is split up in this way. Perhaps this might apply to a SaaS application, but the definition of cloud computing should be much more general

> A cloud computing system can implement any computer program, from data processing to video games. Generally, each application will have its dedicated server.

Reviewer 2: Again, this seems to refer more to cloud computing as a specific application rather than a more general concept.

## Advantages

Cloud computing offers the following advantages to the user:

> - **Scalability:** Adding resources is quicker and easier in a cloud environment then in an on-premises data center which involves purchasing, installing and configuring new hardware.

Reviewer 2: Would mention that this allows access to many more resources, not just that adding resources is quicker and easier.

- **Security:** A cloud solution often provides more security resources than a business, especially a small business, can manage on their own.
- **Availability:** Most cloud solutions have more robust and redundant architecture than a small to medium business can afford, leading to better uptime for users and clients.
- **Performance:** A cloud solution will provide more powerful hardware than a business could afford in its own data center.
- **Cost:** A cloud solution is typically more affordable than an on-premises solution offering the same capabilities.

> Generally, cloud computing removes the necessity of a business to maintain a dedicated data center, with all the hardware and personnel it requires, for a particular solution, allowing the business to allocate those resources to other tasks. Because of this, cloud solutions are increasing in popularity, especially with the increasing cost of keeping hardware and software up to date.

Reviewer 2: Would recommend removing the last sentence.
